### PR TITLE
feat!: Update voucher API

### PR DESCRIPTION
**Pull request by Gozala**
## Overview

Originally [voucher spec](https://hackmd.io/@gozala/w3-accounts) used:

1. `nb.of` (named `nb.product` here) to identify a product.
2. `nb.at` (named `nb.service` here) to identify a provider service of the product.
3. `with` field in `voucher/redeem` was basically web3.storage did so it could self-issued.

With this change we simplify things by:

1. Identifying products via did:key
2. Requiring that `vouche/redeem` used product DID in the `with` field, which in turn implies product owner should delegate capability to the validator.

### Background

I realized that we could simply identify products via did:key, which in turns remove need for both `nb.of` and `nb.at` fields. Not only it simplifies things it also means that any one could create a product (not just us but also our users)  and likewise anyone could install any product as long as product owner delegated that capability by whatever means.
- Of course we may not do anything if some account installs some product we're not aware of, but that is ok it's like `upload/list` we just need to save products in the list product list and only change account settings based on products we're aware of.
- In the future we may also create ability to publishing products, e.g PL could create "pl-tier" product which basically would provide accounts with some space while billing "pl-tier" owner.
    

⚠️ Please note that this branch targets #159 because it fixed bunch of problems which I did not want to deal once again. Once #159 lands I'll change target to main branch

### CODE REVIEW

**Comment by Gozala** [comment]
I'm also starting to think that maybe we don't need `voucher/redeem` but instead it should be something like `product/install`

File: packages/access-api/src/service/voucher-claim.js [hunk 1]
```diff
@@ -6,29 +6,43 @@ import { delegationToString } from '@web3-storage/access/encoding'
  * @param {import('../bindings').RouteContext} ctx
  */
 export function voucherClaimProvider(ctx) {
+  // Provider should have access to delegated vouchers which it can
+  // redelegate. Currently however we just identify "free-tier" with a
+  // provider DID.
+  const products = new Map([
+    [
+      ctx.signer.did(),
```
Comment (alanshaw) [inline packages/access-api/src/service/voucher-claim.js:14]: Should we create a separate DID for this? This is the DID of the service no?

**Review by alanshaw — CHANGES_REQUESTED** [review]

File: packages/access-api/src/service/voucher-claim.js (hunk 1 above)
Comment (Gozala) [reply packages/access-api/src/service/voucher-claim.js:14]: In practice I think we should have DID per product, that said I was not sure what the best way to thread that through, so I thought this was reasonable compromise to get at least discussion started.

**Comment by Gozala** [comment]
After thinking about treading through delegations from voucher -> service I've realized there are bunch of complications:

1. If we rotate service key we need to coordinate all voucher delegations
2. For tests we we'll need to somehow have another set of delegations (because DID in test is different)

Since https://github.com/web3-storage/w3protocol/issues/182 will basically address all the above, I'm going to pause it for now and resume it afterwards

**Pull request closed without merging**